src/main/java/com/musala/atmosphere/commons/websocket/message/Message.java

```diff
@@ -0,0 +1,100 @@
+package com.musala.atmosphere.commons.websocket.message;
+
+import com.musala.atmosphere.commons.RoutingAction;
+
+/**
+ * Represents the abstract WebSocket message properties.
+ *
+ * @author dimcho.nedev
+ *
+ */
+public abstract class Message {
+    protected MessageAction messageAction;
+
+    protected RoutingAction routingAction;
+
+    protected String sessionId;
+
+    protected String agentId;
+
+    public Message(String sessionId, MessageAction messageAction, RoutingAction routingAction) {
```

Comment:
I'd put sessionId as last param. This way the constructors have the message Action first all the time.